gateway-service-knoxtoken/src/main/java/org/apache/knox/gateway/service/knoxtoken/ServletContextWrapper.java

```diff
@@ -0,0 +1,324 @@
+    @Override
+    public ServletContext getContext(String uripath) {
+        return delegate.getContext(uripath);
+    }
+
+    @Override
+    public int getMajorVersion() {
+        return delegate.getMajorVersion();
+    }
+
+    @Override
+    public int getMinorVersion() {
+        return delegate.getMinorVersion();
+    }
+
+    @Override
+    public int getEffectiveMajorVersion() {
+        return delegate.getEffectiveMajorVersion();
+    }
+
+    @Override
+    public int getEffectiveMinorVersion() {
+        return delegate.getEffectiveMinorVersion();
+    }
+
+    @Override
+    public String getMimeType(String file) {
+        return delegate.getMimeType(file);
+    }
+
+    @Override
+    public Set<String> getResourcePaths(String path) {
+        return delegate.getResourcePaths(path);
+    }
+
+    @Override
+    public URL getResource(String path) throws MalformedURLException {
+        return delegate.getResource(path);
+    }
+
+    @Override
+    public InputStream getResourceAsStream(String path) {
+        return delegate.getResourceAsStream(path);
+    }
+
+    @Override
+    public RequestDispatcher getRequestDispatcher(String path) {
+        return delegate.getRequestDispatcher(path);
+    }
+
+    @Override
+    public RequestDispatcher getNamedDispatcher(String name) {
+        return delegate.getNamedDispatcher(name);
+    }
+
+    @Override
+    public Servlet getServlet(String name) throws ServletException {
+        return delegate.getServlet(name);
+    }
+
+    @Override
+    public Enumeration<Servlet> getServlets() {
+        return delegate.getServlets();
+    }
+
+    @Override
+    public Enumeration<String> getServletNames() {
+        return delegate.getServletNames();
+    }
+
+    @Override
+    public void log(String msg) {
+        delegate.log(msg);
+    }
+
+    @Override
+    public void log(Exception exception, String msg) {
+        delegate.log(exception, msg);
+    }
+
+    @Override
+    public void log(String message, Throwable throwable) {
+        delegate.log(message, throwable);
+    }
+
+    @Override
+    public String getRealPath(String path) {
+        return delegate.getRealPath(path);
+    }
+
+    @Override
+    public String getServerInfo() {
+        return delegate.getServerInfo();
+    }
+
+    @Override
+    public String getInitParameter(String name) {
+        String value = delegate.getInitParameter(name);
+        if (value == null) {
+            value =  defaultParams.get(name);
+        }
+        return value;
+    }
+
+    @Override
+    public Enumeration<String> getInitParameterNames() {
+        // Add all keys from the overriddenParams map
+        Set<String> combinedNames = new HashSet<>(defaultParams.keySet());
+
+        // Add all parameter names from the delegate context
+        Enumeration<String> delegateNames = delegate.getInitParameterNames();
+        while (delegateNames.hasMoreElements()) {
+            combinedNames.add(delegateNames.nextElement());
```

Comment:
HashSet eliminates the possibility of duplicate entries, right?

Reply:
I'll add to the test to make sure.